tests/profiler/bob_user_workflow_verbose_profiler_config.yml

```diff
@@ -0,0 +1,216 @@
+      - expectation_type: expect_column_values_to_be_in_set
+        column: $domain.domain_kwargs.column
+        # Parameter builder returns a list of value_sets so take the union() here
+        # TODO: Build `union(sets: List[Set])` method in GE parsing language
+        # TODO: How to reference list of parameters from ParameterBuilder (proposal below)?
+        value_set: union($parameter.my_value_set_for_ids.parameter_list)
+
+  my_rule_for_amounts:
+    # This rule creates expect_column_values_to_be_between & expect_column_mean_to_be_between for each of the _amount column domains
+    # Rationale: I want alerts when my current data varies from prior data (within some bounds)
+    class_name: Rule
+    module_name: great_expectations.profiler.rule.rule
+    domain_builder:
+      # Build list of domains from "amount" column domains: fare_amount,tip_amount,tolls_amount,total_amount
+      class_name: SimpleColumnSuffixDomainBuilder
+      module_name: great_expectations.profiler.domain_builder.simple_column_suffix_domain_builder
+      column_name_suffixes:
+        - _amount
+    parameter_builders:
+      - name: my_mean_parameter_for_batches_in_previous_full_year
+        # Parameter computes the mean of all batches in the previous full year
+
+        # Options for where to calculate aggregate (mean of batch means)
+        # 1. Preferred: MultiBatchParameterBuilder returns a list of parameters calculated on each batch (e.g. [mean_batch_1, mean_batch_2, ...])
+        # 2. Medium complexity configs: (Out of scope for now, but shown here in this example config): MultiBatchAggregateFunctionParameterBuilder which takes a metric_configuration to compute on a list of batches, then "aggregate_fn" parameter which defines an aggregate function to be calculated on all of the computed metrics. (other parameter builders can be customized to calculate non-linear or other more complicated aggregations)
+          # MultiBatchAggregateFunctionParameterBuilder takes optional function, if no function specified then return list of metric values.
+        # 3. High complexity configs: Custom ParameterBuilders / ExpectationConfigurationBuilders can be referenced via plugins
+        # TODO: Streaming mean / stdev consider how would that work?
+        class_name: MultiBatchAggregateFunctionParameterBuilder
+        module_name: great_expectations.profiler.parameter_builder.multi_batch_metric_parameter_builder
+        batch_request:
+          # Filter for batches for previous full year
+          #  using batch_filter_parameters with `year` as one of the parameters
+          #  `year` needs to be defined as a group by the data_connector
+          # TODO: is this possible - how do we specify a data_connector (when calling Profiler.profile()?)?
+          data_connector_query:
+            batch_filter_parameters:
+              year: 2020 # TODO: Can this be relative to the current year? (e.g. specify last year vs the exact year)
```

Comment:
Is the question whether we might allow some parsing language here to allow for parameterization? I think allowing the same language that we use elsewhere in the config would be fine.

In the case of taxi data (and this is maybe the way that the data connectors work for right now??), I believe that year is a string, so being able to specify that we want to convert this to an int, and then operate on it, would be great.

Reply:
Correct - what would be an example of that syntax? `year: max(year)-1` to get the year previous to the latest?

Reply:
I'm not sure I understand how `max` is being used here. I wonder if we'll want a NOW() type of function though.

Reply:
Ah I was thinking `max()` of available years from the data connector (which I'm not sure how we would access), but I also like the idea of a `now()` function. E.g. something more like `year: date_trunc(now(), 'year') - 1`

Reply:
Would it make sense, at least initially, to steer away from complex syntax in YML configuration files, but instead create sophisticated `ParameterBuilder` and `ExpectationConfigurationBuilder` classes to implement the specified use cases? I believe that if we get sufficient experience with this approach, then ideas for higher-level syntax will emerge.

Reply:
I think you are right @alexsherstinsky but this is more in reference to filtering in `BatchRequest`s which I believe can already be done with custom `batch_filters` e.g. `custom_filter_function` (@talagluck would a `custom_filter_function` work for this case?)